docs/Migration-Guide.md

```diff
@@ -0,0 +1,90 @@
+`PAYONE_MERCHANT_ID`            | `MAIN-SHOP_PAYONE_MERCHANT_ID`                             | **Yes**
+`PAYONE_MODE`                   | `MAIN-SHOP_PAYONE_MODE`                                    | **Yes**
+`PAYONE_PORTAL_ID`              | `MAIN-SHOP_PAYONE_PORTAL_ID`                               | **Yes**
+`PAYONE_SUBACC_ID`              | `MAIN-SHOP_PAYONE_SUBACC_ID`                               | **Yes**
+`PAYONE_MODE`                   | `MAIN-SHOP_PAYONE_MODE`                                    | No
+`UPDATE_ORDER_PAYMENT_STATE`    | `MAIN-SHOP_UPDATE_ORDER_PAYMENT_STATE`                     | No
+`SECURE_KEY`                    | `MAIN-SHOP_SECURE_KEY`                                     | No
+
+`SHORT_TIME_FRAME_SCHEDULED_JOB_CRON` and `LONG_TIME_FRAME_SCHEDULED_JOB_CRON` are common to the whole service 
+thus do not require tenant specific configuration.
+
+### 2. Payone portal changes
+
+In Payone go to [Payment portals](https://pmi.pay1.de/merchants/?navi=portal&list=1) page, edit portal with id 
+`PAYONE_PORTAL_ID` and:
+
+Tab        | Field name              | Old value                                                   | New value | Comment
+-----------|-------------------------|-------------------------------------------------------------|-----------------------------------------------------------------------------------------|---------
+_Extended_ | `TransactionStatus URL` | `https://payone-integration.myshop.com/payone/notification` | <code>https://payone-integration.myshop.com/<b>MAIN-SHOP</b>/payone/notification</code> | **Mandatory**
+_General_  | `URL`                   | `https://payone-integration.myshop.com`                     | <code>https://payone-integration.myshop.com/<b>MAIN-SHOP</b>                            | Optional
+
+### 3. Changes in the shops which uses the service
+
+After you setup the service and respective Payone portal - update a shop settings. 
+Find a place where URL request the integration service is configured. 
+You have to change the payment handling URL:
+  1. The old URL expected to be a string containing `/commercetools/handle/payments/`
+  1. Insert your default tenant name (like `MAIN-SHOP`) into this path: <code>**MAIN-SHOP**/commercetools/handle/payments/</code>
+
+### 4. Setup a new tenant (branch,shop,merchant)
+
+  1. Add a new tenant name to `TENANTS`, like `TENANTS=MAIN-SHOP,MERCHANT1`
+  1. [Add a new portal in Payone](https://pmi.pay1.de/merchants/?navi=portal) with `MERCHANT1` path part in the notification URL
+  1. [Add a new commercetools platform account](https://admin.commercetools.com/) 
+  1. In the service settings (environment variables) add the same properties, but with `MERCHANT1_` prefix instead of `MAIN-SHOP_`
+  1. In the new shop configure payment checkout service to connect to the new merchant URL 
+    (like <code>**MERCHANT1**/commercetools/handle/payments/</code>)
+
+### 5. PROFIT
```

Comment:
No _South Park_ Eastern Eggs in the documentation ;(

2b3bd42d1b3d79f00a3f1a31b077e2b573e85ba0